pandas/tests/io/test_pickle.py

```diff
@@ -304,47 +304,39 @@ def test_pickle_v0_15_2():
     tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
 
-class TestPickleCompression(object):
-
-    def setup_class(self):
-        self.path = u'__%s__.pickle' % tm.rands(10)
-
-    def compression_explicit(self, compression):
-        # issue 11666
-        if compression == 'xz':
-            tm._skip_if_no_lzma()
-        with tm.ensure_clean(self.path) as path:
+# ---------------------
+# test pickle compression
+# ---------------------
+def get_random_path():
+    return u'__%s__.pickle' % tm.rands(10)
+
+
+@pytest.mark.parametrize('compression', [None, 'gzip', 'bz2', 'xz'])
+def test_compression_explicit(compression):
+    # issue 11666
+    if compression == 'xz':
+        tm._skip_if_no_lzma()
+    with tm.ensure_clean(get_random_path()) as path:
+        df = tm.makeDataFrame()
+        df.to_pickle(path, compression=compression)
+        df2 = pd.read_pickle(path, compression=compression)
+        tm.assert_frame_equal(df, df2)
+
+
+@pytest.mark.parametrize('compression', ['', 'None', 'bad', '7z'])
+def test_compression_explicit_bad(compression):
+    with tm.assertRaisesRegexp(ValueError,
+                               "Unrecognized compression type"):
+        with tm.ensure_clean(get_random_path()) as path:
             df = tm.makeDataFrame()
             df.to_pickle(path, compression=compression)
-            df2 = pd.read_pickle(path, compression=compression)
-            tm.assert_frame_equal(df, df2)
-
-    def test_compression_explicit(self):
-        compressions = [None, 'gzip', 'bz2', 'xz']
-        for c in compressions:
-            yield self.compression_explicit, c
-
-    def compression_explicit_bad(self, compression):
-        with tm.assertRaisesRegexp(ValueError,
-                                   "Unrecognized compression type"):
-            with tm.ensure_clean(self.path) as path:
-                df = tm.makeDataFrame()
-                df.to_pickle(path, compression=compression)
-
-    def test_compression_explicit_bad(self):
-        compressions = ['', 'None', 'bad', '7z']
-        for c in compressions:
-            yield self.compression_explicit_bad, c
-
-    def compression_infer(self, ext):
-        if ext == '.xz':
-            tm._skip_if_no_lzma()
-        with tm.ensure_clean(self.path + ext) as path:
-            df = tm.makeDataFrame()
-            df.to_pickle(path)
-            tm.assert_frame_equal(df, pd.read_pickle(path))
 
-    def test_compression_infer(self):
-        extensions = ['', '.gz', '.bz2', '.xz', '.no_compress']
-        for ext in extensions:
-            yield self.compression_infer, ext
+
+@pytest.mark.parametrize('ext', ['', '.gz', '.bz2', '.xz', '.no_compress'])
```

Comment:
@jreback, I'm thinking this isn't something that I would advocate for. Here is what `mimetypes` does:

```python
import mimetypes
print(mimetypes.guess_type('file-name.tsv.gz'))
print(mimetypes.guess_type('file-name.tsv.gzip'))
```

outputs:

```
('text/tab-separated-values', 'gzip')
(None, None)
```

So if we use `mimetypes`, the `.gzip` extension won't be recognized. We could code our compression inference to recognize `.gzip` and `.gz`, but I don't think that would be the right decision, since you could then write gzip compressed files that end in `.gzip` using inference. I'd rather stick with the extensions recognized by mimetypes.

If you'd still like to discuss this further via a dedicated issue, just let me know and I'll post it... with the caveat I'm not advocating for it :smirk_cat:.